### ISSUE TITLE
HNC: make HNC work without Stackdriver

### ISSUE THREAD

**Issue by adrianludwin** [kubernetes-sigs/multi-tenancy#770]
I'm not sure why this has stopped working but it would be nice to fix.

**Comment by adrianludwin** [comment]
Wondering if this is related to #777 . I'll have a look at it now.

**Comment by adrianludwin** [comment]
I can't run the container locally via "docker run" either, and there are no logs. Hmm.

**Comment by adrianludwin** [comment]
I do get _some_ output out of 0.2:

```
docker run -it gcr.io/k8s-staging-multitenancy/hnc/controller:v0.2.0
{"level":"error","ts":1591025475.7874,"logger":"controller-runtime.client.config","msg":"unable to get kubeconfig","error":"could not locate a kubeconfig","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/client/config.GetConfigOrDie\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/client/config/config.go:143\nmain.main\n\t/workspace/cmd/manager/main.go:73\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:200"}
```

**Comment by adrianludwin** [comment]
Strangely enough, adding logs to `setupLog.Info()` isn't returning anything, but using `fmt.Println()` works fine. It appears that at least when running locally, it exits when it tries to create the Stackdriver exporter. That could explain why it's working fine on GKE but not other platforms (ugh).

**Comment by adrianludwin** [comment]
Ah - we now start the controller manager quite late, and the `setupLog` doesn't work until it's started. Moving the manager to the top of `main()` gets us logs again, which identifies Stackdriver as the culprit (makes sense).

**Comment by adrianludwin** [comment]
Fixed by #779 

/close
